geonode/layers/views.py

```diff
@@ -507,19 +508,66 @@ def layer_change_poc(request, ids, template='layers/layer_change_poc.html'):
 @login_required
 def layer_list(request):
-    layer = Layer.objects.all()
+    filename = str(datetime.datetime.now()) + "_layers.csv"
     response = HttpResponse(content_type='text/csv')
-    response['Content-Disposition'] = 'attachment; filename="geonode.csv"'
+    response['Content-Disposition'] = 'attachment; filename=' + filename
     writer = csv.writer(response)
+    # Writes header row
+    writer.writerow(['Layer ID','Name',
+                    'Creation Date',
+                    'Title',
+                    'Hazard Set',
+                    'Hazard Type',
+                    'Hazard Period',
+                    'Hazard Unit',
+                    'Region',
+                    'Owner Organization',
+                    'License',
+                    'Restrictions',
+                    'Calculation Method Quality',
+                    'Scientific Quality',
+                    'Layer URL',
+                    'Supplemental Information',
+                    'Data Quality Statement',
+                    'Is Published',
+                    'Metadata Author',
+                    'Owner',
+                    'Keywords',
+                    'Category',
+                    'Abstract'
+                    ])
     for layer in Layer.objects.all():
         regions = layer.regions.all().values('name')
 
         regions = [region['name'] for region in regions]
 
         regions = "||".join(regions)
 
-        writer.writerow([layer.name, layer.title, layer.hazard_type, layer.hazard_period, layer.hazard_unit, settings.SITEURL + layer.get_absolute_url(), unicodedata.normalize('NFKD', layer.supplemental_information).encode('ascii','ignore'), layer.metadata_author.username, layer.owner.username, regions,])
-    return response
+        writer.writerow([layer.id,
+                         layer.name,
+                         layer.creation_date,
+                         layer.title,
+                         layer.hazard_set,
+                         layer.hazard_type,
+                         layer.hazard_period,
+                         layer.hazard_unit,
+                         regions,
+                         layer.owner.organization,
+                         layer.license,
+                         layer.restriction_code_type,
+                         layer.calculation_method_quality,
+                         layer.scientific_quality,
+                         settings.SITEURL + layer.get_absolute_url(),
+                         unicodedata.normalize('NFKD', layer.supplemental_information).encode('ascii','ignore'),
+                         layer.data_quality_statement,
+                         layer.is_published,
+                         layer.metadata_author.username,
+                         layer.owner.username,
+                         layer.keywords,
+                         layer.category,
+                         layer.abstract
+                         ])
+return response
```

Comment:
Same apply here, return statement is not in function. You can check syntax of python files with following command : 
```bash
$ python -m py_compile geonode/layers/views.py
```
This will show you syntax errors like : 
```
  File "geonode/layers/views.py", line 570
    return response
SyntaxError: 'return' outside function
```